docs/proposals/33-machine-image-capabilities.md

```diff
@@ -0,0 +1,546 @@
+}
+```
+
+For each cloud profile the capabilities are defined in the `spec.capabilities` array. The full set of possibilities for each capability is defined here. As some capabilities can have multiple values at the same time an array of possible values is used instead of a single value.
+
+This structure defines the default set of capabilities. If no further information are provided each machine type and machine image will be assigned the default set of capabilities. Thus being compatible with each other.
+
+```yaml
+# CloudProfile Example
+spec:
+  capabilities:
+    hypervisorType: ["gen2", "gen1"]
+    network: ["accelerated", "standard"]
+    storageAccess: ["NVMe", "SCSI"]
+    secureBoot: ["secure", "none"]
+    bootMode: ["uefi-prefered", "uefi", "legacy-bios"]
+    ...
+```
+
+The array of a capability is ordered. The order defines the priority of the values in case of multiple supported images. The first value in the array is the most preferred value. The last value is the least preferred value.
+
+Example: A machine supports hypervisor `gen2` AND `gen1` and an image version offers `gen1` OR `gen2`. Then the image with `gen2` will be preferred.
+
+
+In addition to the default capabilities, the `spec.machineTypes` and the `spec.providerConfig.machineImages.versions` are extended with the `capabilities` structure described above. The image versions in `spec.machineImages.versions` will receive an array of capabilities structures, one entry for each image reference in the providerConfig.
+
+```yaml
+# CloudProfile
+spec:
+  capabilities:
+    architectures: [arm64, amd64]
+    hypervisorType: ["gen2", "gen1"]
+    network: ["accelerated", "standard"] # <-- replaces acceleratedNetworking
+
+  machineImages:
+    - name: gardenlinux
+      updateStrategy: minor
+      versions:
+        # - architectures: [arm64, amd64] # not required anymore
+        #   version: 1592.2.0-gen2
+
+        - architectures: [amd64, arm64]
+          capabilities:
+            - architecture: [arm64]
+              hypervisorType: ["gen2"]
+              network: ["accelerated", "standard"] # <-- not required as its the default
+            - architecture: [amd64]
+              hypervisorType: ["gen2"]
+            - architecture: [amd64]
+              hypervisorType: ["gen1"]
+          classification: supported
+          version: 1592.2.0
+
+  machineTypes:
+    - architecture: amd64
+      cpu: "896"
+      gpu: "0"
+      memory: 12Ti
+      name: Standard_S896om
+      usable: true
+      capabilities:
+        - hypervisorType: ["gen2"] # <-- hypervisorType is overwritten
+
+  providerConfig:
+    machineImages:
+      - name: gardenlinux
+        versions:
+          - architecture: arm64 # <-- architecture is overwritten as only amd64 is supported
+            capabilities:
+              - hypervisorType: ["gen2"]
+              - network: ["accelerated", "standard"] # <-- not required as its the default
```

Comment:
Maybe one should be explicit here and use at least an alias to the default capabilities instead of implicitly assuming (default) capabilities